# Heartmap Index

- Heartmap Index is a community rating index that finds "a place you want to live in for a while, or even stay".
- We analyze whether a community has "a sense of life" from four aspects:
  - `Tastescape Score`Is it good to eat and shop?
  - `Inspiration Index` Can it inspire inspiration?
  - `Romance Quotient` Is it romantic? (Optional)
  - `Night Pulse` Is the night still gentle? (Optional)

In [1]:
import pandas as pd
import numpy as np

# Calculate the final score for each aspect
tastescape_scores = pd.read_csv('data_score/tastescape_scores.csv')
inspiration_index = pd.read_csv('data_score/inspiration_index.csv')

# merge tastescape_scores and inspiration_index

# inspiration_index alse have zip_code
final_scores = tastescape_scores[['zip_code', 'score']].rename(columns={'score': 'tastescape_score'})

# merge inspiration
final_scores = final_scores.merge(
    inspiration_index[['zip_code', 'score']].rename(columns={'score': 'inspiration_index'}),
    on='zip_code',
    how='left'
)

# Fill NaN values with 0 for inspiration_index
final_scores['inspiration_index'] = final_scores['inspiration_index'].fillna(0)

print(final_scores.head(10))


   zip_code  tastescape_score  inspiration_index
0     15222          1.000000           0.473461
1     15237          0.717526           0.000000
2     15212          0.653608           1.000000
3     15146          0.560825           0.000000
4     15213          0.558763           0.817410
5     15205          0.544330           0.314225
6     15206          0.498969           0.745223
7     15108          0.470103           0.000000
8     15217          0.463918           0.651805
9     15219          0.457732           0.847134


In [2]:
final_scores['heartmap_score'] = (
    0.6 * final_scores['tastescape_score'] +
    0.4 * final_scores['inspiration_index']
)
# sort the final scores in descending order
final_scores = final_scores.sort_values(by='heartmap_score', ascending=False)
# reset the index
final_scores = final_scores.reset_index(drop=True)
#print the final scores
print(final_scores[['zip_code', 'heartmap_score']].head(10))

   zip_code  heartmap_score
0     15212        0.792165
1     15222        0.789384
2     15213        0.662222
3     15219        0.613493
4     15206        0.597471
5     15217        0.539072
6     15205        0.452288
7     15237        0.430515
8     15146        0.336495
9     15203        0.334582


### Conclusion

- We can see the best two places are `15212` and `15222`, which are `North Side` and `Downtown` in Pittsburgh, Pennsylvania.
- The second two best places are `15213` and `15219`, which are `Oakland` and `Strip District`
- The last two places are `15206` and `15217`, which are `East Liberty` and `Squirrel Hill`

- I will post the link of the places
  - [North Side](https://zipcodes.org/15212-zip-code/)
  - [Downtown](https://zipcodes.org/15222-zip-code/)
  - [Oakland](https://zipcodes.org/15213-zip-code/)
  - [Strip District](https://zipcodes.org/15219-zip-code/)
  - [East Liberty](https://zipcodes.org/15206-zip-code/)
  - [Squirrel Hill](https://zipcodes.org/15217-zip-code/)